diff --git a/PROVER/Makefile b/PROVER/Makefile
index 976940b..9d8421a 100644
--- a/PROVER/Makefile
+++ b/PROVER/Makefile
@@ -48,7 +48,7 @@ eprover: $(EPROVER)
 	$(LD) -o eprover $(EPROVER) $(LIBS)
 
 eprover-ho: eprover
-	mv eprover eprover-ho
+	cp eprover eprover-ho
 
 E_LTB_RUNNER = e_ltb_runner.o ../lib/CONTROL.a ../lib/HEURISTICS.a\
             ../lib/CLAUSES.a ../lib/ORDERINGS.a ../lib/TERMS.a\
