=== modified file 'src/lightdm-gtk-greeter.c'
--- src/lightdm-gtk-greeter.c	2013-02-09 23:20:39 +0000
+++ src/lightdm-gtk-greeter.c	2013-03-29 12:21:34 +0000
@@ -1273,7 +1273,7 @@
     gtk_container_add (GTK_CONTAINER (menuitem), image);
     gtk_widget_show (GTK_WIDGET (menuitem));
     
-    items = lightdm_get_sessions ();
+    items = lightdm_get_sessions (greeter);
     GSList *sessions = NULL;
     for (item = items; item; item = item->next)
     {
