diff --git a/configure.ac b/configure.ac
index 7a9e097..7fdae11 100644
--- a/configure.ac
+++ b/configure.ac
@@ -123,10 +123,7 @@ AM_CONDITIONAL(USE_X11, test "$USE_X11" = "yes")
 
 dnl Check for VA-API drivers path
 AC_MSG_CHECKING([for VA drivers path])
-LIBVA_DRIVERS_PATH=`$PKG_CONFIG libva --variable driverdir`
-if test -z "$LIBVA_DRIVERS_PATH"; then
-    LIBVA_DRIVERS_PATH="/usr/lib/xorg/modules/drivers"
-fi
+LIBVA_DRIVERS_PATH="/var/run/current-system/sw/lib/xorg/modules/"
 AC_MSG_RESULT([$LIBVA_DRIVERS_PATH])
 AC_SUBST(LIBVA_DRIVERS_PATH)
 
