--- djbfft-0.76.orig/install.c
+++ djbfft-0.76/install.c
@@ -1,3 +1,4 @@
+#include <errno.h>
 #include "substdio.h"
 #include "strerr.h"
 #include "error.h"
--- djbfft-0.76.orig/instcheck.c
+++ djbfft-0.76/instcheck.c
@@ -1,5 +1,6 @@
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <errno.h>
 #include "strerr.h"
 #include "error.h"
 #include "readwrite.h"
--- djbfft-0.76.orig/strerr_sys.c
+++ djbfft-0.76/strerr_sys.c
@@ -1,3 +1,4 @@
+#include <errno.h>
 #include "error.h"
 #include "strerr.h"
 
--- djbfft-0.76.orig/substdi.c
+++ djbfft-0.76/substdi.c
@@ -1,3 +1,4 @@
+#include <errno.h>
 #include "substdio.h"
 #include "byte.h"
 #include "error.h"
--- djbfft-0.76.orig/substdo.c
+++ djbfft-0.76/substdo.c
@@ -1,3 +1,4 @@
+#include <errno.h>
 #include "substdio.h"
 #include "str.h"
 #include "byte.h"
